### PR TITLE
Add snap/bin to /etc/environment

### PR DESCRIPTION
**Pull request by dsame**
# Description
Bug fixing

[snap](https://snapcraft.io/) exists on the base image before the provision has started, so there was no install script for it.

Before https://github.com/actions/virtual-environments/pull/660 the PATH variable in /etc/environment had been inherited from the build environment.

Since https://github.com/actions/virtual-environments/pull/660 the PATH variable /etc/environment it is created from scratch. This became a reason of https://github.com/actions/virtual-environments/issues/753

The inheriting run-time environment from build-time environment does not seem to be a good solution, so the provision script snap-environment.sh to to include `/snap/bin` into run-time PATH is added in this PR. This solves the issue.

The root reason of we need to manipulate with `/etc/environment` is the shell scripts running the user's tasks are launched with `-norc` option and the native profile scripts are ignored.

TODO: should we test the expected run-time PATH is the same as it is constructed in `/etc/environment`?

#### Related issue:
https://github.com/actions/virtual-environments/issues/753

## Check list
- [x] Related issue / work item is attached
- [x] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/linux/scripts/installers/snap-environment.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
```
Comment (al-cheb) [inline images/linux/scripts/installers/snap-environment.sh:1]: Why don't we use configure-environment.sh script?

Comment (dsame) [reply images/linux/scripts/installers/snap-environment.sh:1]: I though about it, but decided the semantic of "Configure environment" does not match exactly the task i am doing: "Complete installation of snap".

The putting the modification of snap to "configure environment" leads to 2 potential problems:

1. the installation of snap continues to be hidden from the provision template
2. there will be inconsistency with the other installed software: why most of them have its own scripts but snap does not

Both problems are insignificant and can be ignored, please ping me if i should use  configure-environment.sh script instead

Comment (miketimofeev) [reply images/linux/scripts/installers/snap-environment.sh:1]: Snap is available on the system by default, we don't need to create additional script
https://snapcraft.io/docs/installing-snapd
image

**Pull request merged**